src/app/src/test/kotlin/org/vaccineimpact/api/tests/logic/BurdenEstimateLogic/CloseBurdenEstimateSetLogicTests.kt

```diff
@@ -0,0 +1,234 @@
+            sut.closeBurdenEstimateSet(setId, groupId, touchstoneVersionId, scenarioId)
+        }.isInstanceOf(MissingRowsError::class.java)
+                .hasMessage("""the following problems occurred:
+Missing rows for AGO, NGA
+For example:
+AGO, age 12, year 2005""")
+    }
+
+    @Test
+    fun `cannot close burden estimate set when responsibility lookup throws an error`()
+    {
+        val writer = mockWriter()
+        Mockito.`when`(writer.isSetEmpty(defaultEstimateSet.id)).doReturn(false)
+
+        val repo = mock<BurdenEstimateRepository> {
+            on { getBurdenEstimateSetForResponsibility(any(), any()) } doReturn defaultEstimateSet
+                    .copy(status = BurdenEstimateSetStatus.PARTIAL)
+            on { getResponsibilityInfo(groupId, touchstoneVersionId, scenarioId) } doThrow
+                    UnknownObjectError(scenarioId, "responsibility")
+            on { getEstimateWriter(defaultEstimateSet) } doReturn writer
+        }
+        val sut = RepositoriesBurdenEstimateLogic(mockGroupRepository(), repo, mockExpectationsRepository(), mock(), mock(), mock())
+
+        Assertions.assertThatThrownBy {
+            sut.closeBurdenEstimateSet(setId, groupId, touchstoneVersionId, scenarioId)
+        }.isInstanceOf(UnknownObjectError::class.java).hasMessageContaining(scenarioId)
+    }
+
+    @Test
+    fun `notifies when a set is marked as complete`()
+    {
+        val repo = mockEstimatesRepository()
+        val mockNotifier = mock<Notifier>()
+        val sut = RepositoriesBurdenEstimateLogic(mockGroupRepository(), repo, mockExpectationsRepository(),
+                mock(), getResponsibilitiesRepository(), mockNotifier)
+
+        sut.closeBurdenEstimateSet(setId, groupId, touchstoneVersionId, scenarioId)
+        verify(mockNotifier).notify("A complete burden estimate set has just been uploaded for $groupId - $disease - $scenarioId")
+    }
+
+    @Test
+    fun `notifies when a set is closed but has missing rows`()
+    {
+        val writer = mockWriter()
+        val mockNotifier = mock<Notifier>()
+        Mockito.`when`(writer.isSetEmpty(defaultEstimateSet.id)).doReturn(false)
+        val repo = mockEstimatesRepository(writer)
+        Mockito.`when`(repo.validateEstimates(any(), any())).doReturn(fakeExpectations.expectedRowLookup())
+        val sut = RepositoriesBurdenEstimateLogic(mockGroupRepository(), repo, mockExpectationsRepository(), mock(), mock(), mockNotifier)
+
+        Assertions.assertThatThrownBy {
+            sut.closeBurdenEstimateSet(setId, groupId, touchstoneVersionId, scenarioId)
+        }.isInstanceOf(MissingRowsError::class.java)
+        verify(repo).changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.INVALID)
+        verify(mockNotifier).notify("A burden estimate set with missing rows has just been uploaded for $groupId - $disease - $scenarioId")
+    }
+
+    @Test
+    fun `notifies when all responsibilities have been fulfilled`()
```

Comment:
These last 2 tests are the new ones; the others have just been moved into their own class